app/Http/Controllers/ForumController.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Repositories\ForumRepositoryEloquent;
+use App\Http\Requests\CreateForumRequest;
+use App\Services\ForumService;
+use App\Http\Resources\ForumResource;
+use App\Http\Requests\UpdateForumRequest;
+
+class ForumController extends Controller
+{
+    protected $forumRepository;
+
+    public function __construct(ForumRepositoryEloquent $forumRepository)
+    {
+        $this->forumRepository = $forumRepository;
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index(Request $request)
+    {
+        if ($request->has('topic')) {
+            $topic = $this->forumRepository->findWhere(['topic' => $request->topic]);
+        } else {
+            $topic = $this->forumRepository->all();
+        }
+
+        return ForumResource::collection($topic);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(CreateForumRequest $request, ForumService $forumService)
+    {
+        $data = $forumService->create($request);
```

Comment:
biar seragam dengan yang lain, gimana kalau dibuat jadi gini ?
$data = $forumService->create($request);
$forum = $this->forumRepository->create($data)
        return new ForumResource($forum);